actionview/test/actionpack/controller/render_test.rb

```diff
@@ -785,6 +790,18 @@ def test_simple_show
     assert_equal "<html>Hello world!</html>", @response.body
   end
 
+  def test_controller_does_not_leak_instance_variables
+    get :render_instance_variables
+
+    assert_response 200
+    assert_equal <<~BODY.squish, @response.body
+      [:@_assigns, :@_config, :@_controller, :@_default_form_builder,
+        :@_ivars, :@_request, :@current_template, :@lookup_context,
+        :@output_buffer, :@variable_for_layout, :@view_flow, :@view_renderer,
+        :@virtual_path]
+    BODY
+  end
```

Comment:
Since you went through the trouble of verifying all of these variables in https://github.com/rails/rails/pull/49858#issuecomment-1786350525, would it be worth adding comments here?  Something like:

```ruby
def test_controller_does_not_leak_instance_variables
  expected = [
    @_assigns, # attr_internal on ActionView::Base
    @_config, # attr_internal on ActionView::Base
    @_controller, # attr_internal on ActionView::Helpers::ControllerHelper
    ...
  ].sort.inspect

  get :render_instance_variables

  assert_response 200
  assert_equal expected, @response.body
end
```

Reply:
I don't love the pattern of comments in tests, if we're going to add these comments I'd rather them be in the code.

Reply:
My thinking is that if future changes break this test, whoever is fixing it will immediately see these comments.  Also, I'm not sure of a single place to put such comments in the actual code.

Reply:
Ah ok, I thought it was more related to something you'd need to know outside of the tests. I'm fine with adding the comments in this case.